src/components/avatar/index.js

```diff
@@ -142,28 +156,48 @@ export default class Avatar extends BaseComponent {
 
   getStatusBadgeColor(status) {
     switch (status) {
-      case Avatar.modes.NONE:
-        return null;
+
       case Avatar.modes.AWAY:
         return Colors.yellow30;
       case Avatar.modes.ONLINE:
         return Colors.green30;
       case Avatar.modes.OFFLINE:
         return Colors.dark60;
+      case Avatar.modes.NONE:
       default:
         return null;
     }
   }
 
-  getBadgeColor(isOnline, status) {
-    const onlineOverride = status === STATUS_MODES.NONE ? isOnline : false;
-    const badgeColor = onlineOverride ? Colors.green30 : this.getStatusBadgeColor(status);
-    return badgeColor;
+  getBadgeColor() {
+    const {isOnline, status} = this.props;
+    const statusColor = this.getStatusBadgeColor(status);
+    const onlineColor = isOnline ? Colors.green30 : undefined;
+
+    return _.get(this.props, 'badgeProps.backgroundColor') || statusColor || onlineColor;
   }
-  
+
   getBadgeSize = () => _.get(this.props, 'badgeProps.size', DEFAULT_BADGE_SIZE);
-  
-  getBadgeBorderWidth = () => _.get(this.props, 'badgeProps.borderWidth', DEFAULT_BADGE_BORDER_WIDTH);
+
+  getBadgeBorderWidth() {
+    const {badgeProps} = this.props;
+    const borderWidths = _.get(this.props, 'badgeProps.borderWidth', DEFAULT_BADGE_BORDER_WIDTH);
+
+    if (_.get(this.props, 'badgeProps.icon')) {
+      return badgeProps.borderWidth || 0;
```

Comment:
you've extracted the `borderWidth` couple of lines before, why not use it here? 
maybe I don't understand the logic

Reply:
all badge types except icon badge should DEFAULT_BADGE_BORDER_WIDTH. So for Icon badge is you don't pass borderWidth you should get 0. Why not undefined || null. I use border width for calculating badge location on Avatar. so undefined is out of the question. Null seemed less "safe" than 0

Reply:
it feels like this logic should move to our private repo, no? it's all very specific to our product's use case unless it feels like good UX guidelines we should enforce.